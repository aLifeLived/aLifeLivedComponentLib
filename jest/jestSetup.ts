/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
