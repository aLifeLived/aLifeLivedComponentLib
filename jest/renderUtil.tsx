import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
