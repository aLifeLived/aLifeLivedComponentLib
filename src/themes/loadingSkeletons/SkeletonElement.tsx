import {
  backgroundColor,
  BackgroundColorProps,
  border,
  BorderProps,
  color,
  ColorProps,
  createRestyleComponent,
  createVariant,
  layout,
  LayoutProps,
  spacing,
  SpacingProps,
  VariantProps,
} from '@shopify/restyle';
import { Box } from '../box/Box';
import { Theme } from '../theme';

export const skeletonElementsVariants = {
  text: {
    width: '70%',
    height: 10,
    backgroundColor: 'backgroundContrastMid',
  },
  title: {
    width: '50%',
    height: 20,
    backgroundColor: 'backgroundContrastMid',
  },
  avatar: {
    width: 50,
    height: 50,
    borderRadius: 'xXlarge',
    borderColor: 'backgroundContrastMid',
    borderWidth: 1,
  },
  thumbNail: {
    width: 50,
    height: 50,
  },
};

export type StyledSkeletonElementProps = SpacingProps<Theme> &
  LayoutProps<Theme> &
  BorderProps<Theme> &
  BackgroundColorProps<Theme> &
  ColorProps<Theme> &
  VariantProps<Theme, 'skeletonElementsVariants'>;

export const SkeletonElement = createRestyleComponent<
  StyledSkeletonElementProps,
  Theme
>(
  [
    spacing,
    layout,
    border,
    backgroundColor,
    color,
    createVariant({ themeKey: 'skeletonElementsVariants' }),
  ],
  Box
);
