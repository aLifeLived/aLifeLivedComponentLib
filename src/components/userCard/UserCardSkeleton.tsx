import React from 'react';

// Components
import { Box } from '../../themes/box/Box';
import { SkeletonElement } from '../../themes/loadingSkeletons/SkeletonElement';
import { SystemAnimatableBox } from '../../themes/systemAnimatableBox/SystemAnimatableBox';

type UserCardSkeleton = {
  testID?: string;
};

export const UserCardSkeleton: React.FC<UserCardSkeleton> = ({ testID }) => {
  return (
    <SystemAnimatableBox
      flexDirection="row"
      animation="flash"
      iterationCount="infinite"
      alignItems="center"
      useNativeDriver={true}
    >
      <SkeletonElement variant="avatar" />
      <Box width="100%" marginLeft="s" testID={testID}>
        <SkeletonElement variant="title" />
      </Box>
    </SystemAnimatableBox>
  );
};
