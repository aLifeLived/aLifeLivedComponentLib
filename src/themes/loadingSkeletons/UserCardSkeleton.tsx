import React from 'react';

// Components
import { Box } from '../box/Box';
import { SkeletonElement } from './SkeletonElement';
import { SystemAnimatableBox } from '../systemAnimatableBox/SystemAnimatableBox';

export const UserCardSkeleton = () => {
  return (
    <SystemAnimatableBox
      flexDirection="row"
      alignContent="center"
      alignItems="center"
      animation="flash"
      iterationCount="infinite"
      useNativeDriver={true}
    >
      <SkeletonElement variant="avatar" />
      <Box flexDirection="column" width="100%" marginLeft="s">
        <SkeletonElement variant="title" />
        <SkeletonElement variant="text" />
      </Box>
    </SystemAnimatableBox>
  );
};
