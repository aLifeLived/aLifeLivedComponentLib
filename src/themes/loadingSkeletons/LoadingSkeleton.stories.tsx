import React from 'react';
import { storiesOf } from '@storybook/react-native';

// Component
import { UserCardSkeleton } from './UserCardSkeleton';
import { Box } from '../box/Box';

storiesOf('Loading Skeletons', module)
  .addDecorator(Story => Story())
  .add('User card', () => (
    <Box margin="m">
      <UserCardSkeleton />
    </Box>
  ));
