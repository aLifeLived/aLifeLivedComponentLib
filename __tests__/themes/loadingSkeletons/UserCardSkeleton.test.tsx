import React from 'react';
import { render } from '../../../jest/renderUtil';

// Component
import { UserCardSkeleton } from '../../../src/themes/loadingSkeletons/UserCardSkeleton';

const setup = () => {
  const utils = render(<UserCardSkeleton />);

  return {
    ...utils,
  };
};

describe('themes >> Loading Skeleton >> UserCardSkeleton', () => {
  describe('Given User card', () => {
    test('Should match snapshot', () => {
      const { toJSON } = setup();
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
