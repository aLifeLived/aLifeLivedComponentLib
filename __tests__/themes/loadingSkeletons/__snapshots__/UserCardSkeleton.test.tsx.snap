// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes >> Loading Skeleton >> UserCardSkeleton Given User card Should match snapshot 1`] = `
<View
  forwardedRef={[Function]}
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
        },
        Array [
          Object {},
          Array [
            Object {
              "borderColor": "#eaeaea",
              "borderRadius": 48,
              "borderWidth": 1,
              "height": 50,
              "width": 50,
            },
          ],
        ],
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "column",
          "marginLeft": 8,
          "width": "100%",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
          },
          Array [
            Object {},
            Array [
              Object {
                "backgroundColor": "#eaeaea",
                "height": 20,
                "marginBottom": 16,
                "width": "50%",
              },
            ],
          ],
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "display": "flex",
          },
          Array [
            Object {},
            Array [
              Object {
                "backgroundColor": "#eaeaea",
                "height": 10,
                "width": "70%",
              },
            ],
          ],
        ]
      }
    />
  </View>
</View>
`;
